src/main/scala/org/broadinstitute/dsde/workbench/sam/util/ShadowRunner.scala

```diff
@@ -21,18 +22,33 @@ object ShadowRunner {
 }
 
 trait ShadowRunner {
-  val clock: Clock[IO]
+  val timer: Timer[IO]
+  lazy val clock: Clock[IO] = timer.clock
   val resultReporter: ShadowResultReporter
-  implicit val contextShift: ContextShift[IO]
+  val realContextShift: ContextShift[IO]
+  val shadowContextShift: ContextShift[IO]
+
+  val shadowTimeout = 5 seconds
 
   protected def runWithShadow[T](methodCallInfo: MethodCallInfo, real: IO[T], shadow: IO[T]): IO[T] = {
     for {
-      realTimedResult <- measure(real)
-      _ <- contextShift.evalOn(ShadowRunner.shadowExecutionContext)(measure(shadow).runAsync {
+      // start both real and shadow concurrently
+      realTimedResultAsync <- measure(real).start(realContextShift)
+      shadowTimedResultAsync <- measure(shadow).start(shadowContextShift)
+
+      // wait for real
+      realTimedResult <- realTimedResultAsync.join
+      // wait for shadow for up to shadowTimeout beyond real
+      shadowTimedResult <- IO.race(shadowTimedResultAsync.join, timer.sleep(shadowTimeout))(shadowContextShift).map {
```

Comment:
Looks like `IO.race` cancels the slower IO. I remember you were considering letting the shadow keep running even if it took longer than the timeout to help prevent desynchronization, but not sure what you ended up deciding